avahi-autoipd/main.c

```diff
@@ -1218,10 +1236,36 @@ static int loop(int iface, uint32_t addr) {
 
                 if (info.sender_ip_address == addr) {
 
+                    /* Normal conflict? */
                     if (memcmp(hw_address, info.sender_hw_address, ETHER_ADDRLEN)) {
-                        /* Normal conflict */
-                        conflict = 1;
-                        daemon_log(LOG_INFO, "Received conflicting normal ARP packet.");
+                        int should_defend = (state == STATE_RUNNING);
+
+                        /* Have we seen any other conflicting ARP packets within the last DEFEND_INTERVAL seconds? */
+                        if ((should_defend) && (has_timestamp(&last_conflict))) {
+                            /* Check if the time recorded for the previous conflicting ARP packet is recent */
+                            usec = avahi_age(&last_conflict) / 1000ULL;
+                            daemon_log(LOG_DEBUG, "Previous conflict packet received %llu ms  ago", usec);
```

Comment:
No it won't. As soon as we receive a 2nd ARP conflict packet, we will give up our current IP address. None of the debug messages appear to have timestamps, this is most likely because they get sent to syslog (timestamps will be added there)